pandas-stubs/core/frame.pyi

```diff
@@ -36,7 +40,7 @@ _ErrorType = Literal["raise", "ignore"]
 
 _ListLike = Union[Series, Index, _np.ndarray, Sequence]
 
-_ColSubsetType = Union[Series, DataFrame, List[_str], _str, _np.ndarray[_np.str_]]
+_ColSubsetType = Union[Series, "DataFrame", List[_str], _str, _np.ndarray[_np.str_]]
```

Comment:
Quoting DataFrame shouldn't be needed. Mypy does multiple passes of the stub files and can manage undefined references.